todoapp.py

```diff
@@ -0,0 +1,153 @@
+import sqlite3
+
+# from collections import namedtuple
+
+from flask import Flask, request, render_template, jsonify
+from sqlite3 import Error
+
+
+# Row = namedtuple('Row', 'id, title, category, expired_date, '
+#                         'is_done, completed_date')
+
+app = Flask(__name__)
+# DB = 'todo.db'
+DB = 'test.db'
+
+
+@app.route('/')
+@app.route('/home/')
+def home():
+    return render_template('home.html', title='Home')
+
+
+@app.route('/api/tasks/', methods=['GET'])
+def show_tasks():
+    args = request.args
+    if args:
+        conditions = [f'{key} = "{args[key]}"' for key in args]
+        conditions = f'WHERE {" AND ".join(conditions)}'
+    else:
+        conditions = ''
+    fields = 'id, title, category, expired_date, is_done, completed_date'
+    query = f'SELECT {fields} FROM task {conditions}'
+    # print(query)
+
+    con = sqlite3.connect(DB)
+    cur = con.cursor()
+    try:
+        cur.execute(query)
+        rows = cur.fetchall()
+    except Error as e:
+        # print(e)
+        return f'Error <{e}> occurred during select data from database', 400
+    finally:
+        cur.close()
+        con.close()
+
+    if not rows:
+        return "Not found tasks according query", 404
```

Comment:
For your API, we selected JSON format for a response (using `jsonify`), but in case of some errors, we return just string. It will better if you be consistent and return some general JSON response in case of error too. Just for example: ``` return {"success": False, "error": "Not found tasks"}, 404 ```

Reply:
Done, but now a have some other questions.
1. If i use key `success`  rhen logically i need use the same key in the success case (may be something as `{'success': True, 'data': jsonify(tasks)}`), so we can check if the request was finished with success or not.  On the other hand the key `error` can be checked for the same purpose, and the key `success' may be overkill.
2. Is it nesessary that all handlers return the same data type in the body of response (in my case - json), or it is permissible that other handlers return other data type in the body of response (string for example)?  Or this rule applies only for the returned data type from one handler?

Reply:
Yes, good questions from you. In this case, the `success` field in response really looks overkill, because our HTTP status shows us that's wrong or not. This kind of question about RESTful API, you can read some tips and best practices [here ](https://florimond.dev/blog/articles/2018/08/restful-api-design-13-best-practices-to-make-your-users-happy/ )and [here](https://www.vinaysahni.com/best-practices-for-a-pragmatic-restful-api).